Userland/Libraries/LibGfx/Bitmap.cpp

```diff
@@ -389,66 +389,114 @@ ErrorOr<NonnullRefPtr<Gfx::Bitmap>> Bitmap::scaled(float sx, float sy) const
-    auto new_bottom_y = new_height - 1;
-    for (int x = 0; x < new_width - 1; x++) {
-        auto p = static_cast<float>(x) * static_cast<float>(old_width - 1) / static_cast<float>(new_width - 1);
 
-        int i = floorf(p);
-        float u = p - static_cast<float>(i);
+        // Right strip (excluding last pixel)
+        auto old_right_x = old_width - 1;
+        auto new_right_x = new_width - 1;
+        for (int y = 0; y < new_height - 1; y++) {
+            auto q = static_cast<float>(y) * static_cast<float>(old_height - 1) / static_cast<float>(new_height - 1);
 
-        auto a = get_pixel(i, old_bottom_y);
-        auto b = get_pixel(i + 1, old_bottom_y);
-        auto color = a.interpolate(b, u);
-        new_bitmap->set_pixel(x, new_bottom_y, color);
-    }
+            int j = floorf(q);
+            float v = q - static_cast<float>(j);
 
-    // Right strip (excluding last pixel)
-    auto old_right_x = old_width - 1;
-    auto new_right_x = new_width - 1;
-    for (int y = 0; y < new_height - 1; y++) {
-        auto q = static_cast<float>(y) * static_cast<float>(old_height - 1) / static_cast<float>(new_height - 1);
+            auto c = get_pixel(old_right_x, j);
+            auto d = get_pixel(old_right_x, j + 1);
 
-        int j = floorf(q);
-        float v = q - static_cast<float>(j);
+            auto color = c.interpolate(d, v);
+            new_bitmap->set_pixel(new_right_x, y, color);
+        }
 
-        auto c = get_pixel(old_right_x, j);
-        auto d = get_pixel(old_right_x, j + 1);
+        // Bottom-right pixel
+        new_bitmap->set_pixel(new_width - 1, new_height - 1, get_pixel(physical_width() - 1, physical_height() - 1));
+        return new_bitmap;
+    } else if (old_height == 1) {
+        // Copy horizontal strip multiple times (excluding last pixel to out of bounds).
+        auto old_bottom_y = old_height - 1;
+        for (int x = 0; x < new_width - 1; x++) {
+            auto p = static_cast<float>(x) * static_cast<float>(old_width - 1) / static_cast<float>(new_width - 1);
+            int i = floorf(p);
+            float u = p - static_cast<float>(i);
 
-        auto color = c.interpolate(d, v);
-        new_bitmap->set_pixel(new_right_x, y, color);
-    }
+            auto a = get_pixel(i, old_bottom_y);
+            auto b = get_pixel(i + 1, old_bottom_y);
+            auto color = a.interpolate(b, u);
+            for (int new_bottom_y = 0; new_bottom_y < new_height; new_bottom_y++) {
+                // Interpolate color only once and then copy into all columns.
+                new_bitmap->set_pixel(x, new_bottom_y, color);
+            }
+        }
+        for (int new_bottom_y = 0; new_bottom_y < new_height; new_bottom_y++) {
+            // Copy last pixel of horizontal strip
+            new_bitmap->set_pixel(new_width - 1, new_bottom_y, get_pixel(physical_width() - 1, old_bottom_y));
+        }
+        return new_bitmap;
+    } else if (old_width == 1) {
```

Comment:
Nit: since we `return` on the line above, this could be an `if` rather than an `else if`.